### PR TITLE
Repurpose build  setting for optional jvm memory

### PR DESCRIPTION
**Pull request by paololim**

### CODE REVIEW

File: spec/delta-config.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
 	{ "name": "dockerfile", "type": "string" },
 	{ "name": "initial_number_instances", "type": "long", "description": "When first deploying this branch, the number of instances we create" },
 	{ "name": "instance_type", "type": "instance_type" },
-	{ "name": "memory", "type": "long", "description": "The number of MiB of memory to reserve for the container." },
+	{ "name": "memory", "type": "long", "required": false, "description": "The number of MiB of memory to reserve for the container." },
```
Comment (ninjabem) [inline spec/delta-config.json]: Does the description of this need to change? Looks like "memory" is now used for setting JVM heap?

Comment (ninjabem) [reply spec/delta-config.json]: And does repurposing this field affect any deployed apps?

Comment (paololim) [reply spec/delta-config.json]: repurposing should not affect other apps right now - it really is not being used and we have removed it from everywhere.

Comment (paololim) [reply spec/delta-config.json]: let me update the desription

**Pull request merged**